lib/beaker/host_prebuilt_steps.rb

```diff
@@ -336,7 +336,8 @@ def copy_ssh_to_root host, opts
     def hack_etc_hosts hosts, opts
       etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
       hosts.each do |host|
-        etc_hosts += "#{host['vm_ip'] || host['ip'].to_s}\t#{host[:vmhostname] || host.name}\n"
+        domain = get_domain_name(host)
+        etc_hosts += "#{host['vm_ip'] || host['ip'].to_s}\t#{host[:vmhostname] || host.name}.#{domain} #{host[:vmhostname] || host.name}\n"
```

Comment:
Maybe in the process I would go a step further and improve readability here a notch:

``` ruby
  ip = host['vm_ip'] || host['ip'].to_s
  hostname = host[:vmhostname] || host.name
  etc_hosts += "#{ip}\t#{hostname}.#{domain} #{hostname}\n"
```